### PR TITLE
Custom product data fields should be a feature flag data model

### PR DESCRIPTION
**Pull request by gherceg**
## Summary

https://dimagi-dev.atlassian.net/browse/SS-188

Custom product data fields are only accessible when the `COMMTRACK` feature flag is enabled. On the linked projects page, this data model currently shows up as a generally accessible data model which is not the case. This change moves the data model to `FEATURE_FALG_DATA_MODELS` which means this will only appear when the corresponding feature flag is enabled. Screenshots below.

With feature flag disabled:
Screen Shot 2021-07-28 at 11 07 26 AM
Screen Shot 2021-07-28 at 10 59 20 AM

With feature flag enabled:
Screen Shot 2021-07-28 at 11 08 48 AM
Screen Shot 2021-07-28 at 11 11 05 AM

## Feature Flag

`LINKED_DOMAINS` and `COMMTRACK`
## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

Good test coverage in this area. Made adequate changes to existing tests to account for this change.
### QA Plan

No QA.
### Safety story

Locally tested.
### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by orangejenny — APPROVED** [review]

**Comment by gherceg** [comment]
FYI I'm aware there is a missing migration. Holding off on this PR until some other PRs are merged into master that contain migrations in this same area.

**Comment by gherceg** [comment]
Change of plans: created the migration here and will merge once the build passes, then update the expected conflicts between branches on staging

**Pull request merged**
